content/blog/findings-when-coverting-tests-to-pester5.md

```diff
@@ -0,0 +1,303 @@
+contribution (the tests added or changed in the contribution) and the
+new contribution together with the existing code.
+
+Pester is about to be released in a new version 5.0.0 with breaking changes
+so I set out to see what was needed to convert our existing tests that are
+run by Pester 4.x. See the release note under [Pester Releases](https://github.com/pester/Pester/releases)
+where the RC6 release notes contain documentation needed for converting.
+
+Pester is primarily maintained and developed by [Jakub Jareš](https://github.com/nohwnd).
+Pester 5 was able to be realized thanks to the hard work by Jakub during
+many weekends. If you have the means then consider [sponsoring Jakub](https://github.com/sponsors/nohwnd)
+or [sponsoring Pester](https://opencollective.com/pester) to show the
+appreciation we have for Jakub's dedication to the community to help us
+make this OSS tool better.
+
+## Tests
+
+I thought I start with the common module SqlServerDsc.Common that contains
+helper function for the DSC resource in the SqlServerDsc module. I knew it
+used a lot of different techniques and also had older tests since pre-Pester 4.
+
+After I started looking at the test I realized that the tests had been refactor
+over the years so they pretty much used modern Pester techniques by having
+setup and teardown in `Before*`- and `After*`-blocks mostly through out.
+So converting tests that do not use this will take considerably longer that
+it took me to run convert these test. Converting 175 tests (~3000 lines of
+code) to ~8 hours.
+
+Even though the tests used the modern Pester techniques by having mocks
+and other test preparation (setup) and teardown in `Before*`- and `After*`-blocks
+don't expect the tests to just run under Pester 5.
+
+## Converting
+
+It took, and will take some time, to learn new practices around how Pester
+does Discovery and Run which is new to Pester 5, especially around
+`InModuleScope`-block.
+
+Pester 5 is mostly backwards compatible with the syntax for cmdlets used
+with Pester 4. But recommended it to use the the names of the cmdlets as
+some cmdlets in Pester 4 are now aliases in Pester 5.
+
+But what took most time was to understand the error messages that was
+outputted when running test that was not "Pester 5 compatible". When the
+tests was wrongly written for Pester 5 it took some time to understand if
+the error was from the Pester, bug in tests, or bug in the code being
+tested. 99.9% of the time it was the tests that needed converting.
+
+> Converting to Pester 5 did help found a bug in the tests that we had
+> missed with Pester 4. There was a need for mocking a line in the code
+> that messed up the PowerShell session. That needed a change in the code
+> being testing.
+
+After writing Pester tests in Pester 3 and 4 I gotten used to some error
+messages returned by Pester when messing up tests, the same will happen
+for Pester 5 pretty quick I think.
```

Comment:
"for Pester 5, eventually." That would emphasize that it's slightly different and need to get used to.